docs/source/common/optimizers.rst

```diff
@@ -67,28 +78,89 @@ Here is the same example as above using a ``closure``.
             opt.zero_grad()
 
 
-.. code-block:: python
+.. tip:: Be careful where you call ``zero_grad`` or your model won't converge. It is good pratice to call ``zero_grad`` before ``manual_backward``.
+
+
+.. testcode:: python
+
+    import torch
+    from torch import Tensor
+    from pytorch_lightning import LightningModule
+
+    class SimpleGAN(LightningModule):
+
+        def __init__(self):
+            super().__init__()
+            self.G = Generator()
+            self.D = Discriminator()
+
+            # Important: This property activate ``manual optimization`` for this model
+            self.automatic_optimization = False
+
+        def generator_loss(self, d_z: Tensor) -> Tensor:
+            # the closer ``d_z`` is from 1,
+            # the better the generator is able to fool the discriminator
+            return -1 * torch.log(d_z).mean()
+
+        def discriminator_loss(self, d_x: Tensor, d_z: Tensor) -> Tensor:
+            # the closer is ``d_x`` from 1 and ``d_z`` from 0,
+            # the better the discriminator is able to distinguish
+            # true data from generated ones
+            return -1 * (torch.log(d_x).mean() + torch.log(1 - d_z).mean())
+
+        def sample_z(self, n) -> Tensor:
+            sample = self._Z.sample((n,))
+            return sample
+
+        def sample_G(self, n) -> Tensor:
+            z = self.sample_z(n)
+            return self.G(z)
+
+        def training_step(self, batch, batch_idx, optimizer_idx, *args):
+            # Implementation follows https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html
+            g_opt, d_opt = self.optimizers()
```

Comment:
Maybe little comment about the unused optimiser_idx arg?

Reply:
Removed entirely.